### PR TITLE
[PoC] Removing duplicate code from controllers and views that deal with tabs

### PR DESCRIPTION
**Pull request by Lucasnar**
Work related to https://github.com/poanetwork/blockscout/issues/748 and https://github.com/poanetwork/blockscout/issues/706.

There are three commits on this PR, each one of them provides a way we could address this duplication issue.

The first commit (https://github.com/poanetwork/blockscout/commit/be020116d61763173f1ea3c880601476ae25aa4b) is intended to validate a solution using Ajax. It is a work on the `Address` page for the `Transactions` tab.

The second commit (https://github.com/poanetwork/blockscout/commit/8a4f8504d991346488bdad1e87cd4f340c2a078f) is intended to validate a solution that does not use Ajax, but centralizes all controllers into one (with one main action). It is a work on the `Transaction` page, for all tabs (`Token Transfers`, `Transactions` and `Logs`).

The third commit (https://github.com/poanetwork/blockscout/pull/765/commits/313b549268d20fefac93b0dc3476438e284f510d) is an alternative approach suggested in an internal discussion we had. It is intended to validate a solution that also does not use Ajax, and is simpler than the one in the second commit, but keeps a little bit of necessary duplication.

All the proposed solutions solve the duplication issue, but each of them will have advantages over the other:

|                 First Solution (Ajax)                 |         Second Solution (One controller)         | Third Solution  (one module to concentrate the common params) |
|:-----------------------------------------------------:|:------------------------------------------------:|---------------------------------------------------------------|
|             Page loads faster (better UX)             |  Faster to implement right now on all pages/tabs | Even faster to implement right now                            |
|               Overview loads before tab               | Easier to understand code (less places to touch) | Easier to understand code (less places to touch)              |
| Do not load overview every time an user changes a tab |   Keeps no duplication in views and controllers  |                                                               |

The drawbacks I see from the proposed solutions are:

|          First Solution (Ajax)          |             Second Solution (One controller)             | Third Solution  (one module to concentrate the common params) |
|:---------------------------------------:|:--------------------------------------------------------:|---------------------------------------------------------------|
|           Harder to implement           |   The controller that handles all tabs may get too big   | Some code will still end up duplicated*                       |
| Harder to understand for new developers | The overview will be loaded every time a tab is clicked. | The overview will be loaded every time a tab is clicked.      |

_* In this third solution, the code that gets the transaction and passes it to the assigns will end up duplicated; also, all views will load the overview and tabs partials. It is not a huge code duplication (just a couple of lines for each controller and view) and this could be even considered good since it keeps the controllers separated, as opposed than everything in a single controller, like the second proposed solution._

There surely are other solutions to this issue. It would be awesome if you guys could take a look at what is done here and comment on the solution you think would be the best for the project; of course, feel free to suggest another one and changes to the solutions presented here. Note that there may be some code here that could be better, but for this PoC we did not wanted to spend much time on letting the code pretty, since it is just to prove a point. After the discussion here, we will open subsequent PRs to deal with all the tabs properly.

### CODE REVIEW

**Comment by Lucasnar** [comment]
After a long discussion about this done personally, we decided to go with the third option (represented by the third commit). It is the simplest approach to implement right now and solves the controller duplication issue. About the views' duplicated code, we will do something similar to what is done in the second commit.

The third commit also is more expressive and won't add complexity compared to the code we currently have; that is, new contributors won't have to study and get used to a very different architecture than the one Phoenix provides. With the first proposed solution, we would have to control a lot of routes through Javascript and think of a way to organize all the code of the tabs (a kinda huge architecture decision); this increases the project complexity and we would have to spend a lot of time testing different approaches in order to make a decision that would lead to the best solution. With the second proposed solution, we would be changing the way Phoenix does routing for some controllers since they would be responsible to route the request to the correct tab; also, as more tabs are needed, the controller grows and it can become huge, which is a problem that would need more architectural decisions to solve.

The first proposed solution is still the one that provides the best UX at the moment, and we considered that. With the third proposed solution, we will not generate a huge diff and impact on the project, but we will solve the duplication issue. This makes it easier to migrate to a better JS/Ajax solution in the future, with the best UX.

In conclusion, I will close this PR and the related PoC issue (https://github.com/poanetwork/blockscout/issues/748), and update the original issue (https://github.com/poanetwork/blockscout/issues/706) with the next steps.
